### ISSUE TITLE
[Bug] Vanced preferred Video Resolution & Override Resolution (Video settings) doesn't work with latest YT changes.

### ISSUE THREAD

**Issue by Harmonie666** [YTVanced/Vanced#697]
**Bug description**
"We don't really like the new settings since they introduce an extra step to change the video resolution" 
– I don't like it either! YT just released another useless change. Video quality is now capped at 720p for every video, even if you have unlimited mobile data, or high speed WiFi.
https://www.gsmarena.com/youtube_android_ios_mobile_app_new_video_resolution_settings-news-48821.php

"Lastly, there is the Advanced option that lets you select a specific resolution you want the video to play in and you can set the resolution as high as 4K. Unfortunately, this last option needs to be activated for each video individually — you can't choose to play all videos at a specific resolution by default."
https://www.androidpolice.com/2021/04/23/youtube-mobile-app-gets-more-exhaustive-video-quality-controls/

So basically this new change by YT, made Vanced's preferred video resolution & Override Video Resolution settings useless. They don't work anymore. Enabled 4k resolution in Vanced video settings (for example when connected to WiFi)  doesn't work anymore. Or choosing 1080p for Mobile data also doesn't work anymore, since every video is automatically capped at 720p. **Sometimes it even goes back to 480p.**

Reddit post:
https://www.reddit.com/r/Vanced/comments/mz8gw3/we_dont_really_like_the_new_settings_since_they/?utm_medium=android_app&utm_source=share

**Variant**
Non-Root

**Vanced version**
16.02.35 - Build - 01.47.00

**Android version**
Android 10

**Device**
ROG Phone 2

**Comment by Harmonie666** [comment]
Also check this:
https://www.reddit.com/r/Vanced/comments/mz4oga/suggestion_for_new_version_161434_please_fix_this/?utm_medium=android_app&utm_source=share

**Comment by Jarvis-Ank** [comment]
I am facing the same problem as the "Video quality preference" and "Video settings" in Vanced settings doesn't match.. I've set default resolution for both wifi and mobile to 360p. Then in Video quality preference switched to data saver.. still every time I play a video it resolution is caping at either 1080p or above.. I've just updated to the latest version yesterday..

Device 
Realme 6, Non Root, Android 11
Vanced version: 16.14.34

**Comment by WarrenWillingham** [comment]
Same issue here. Video quality is all over the place after youtube's resolutions change
Oneplus 6t  non-root
Can provide more info if needed

**Comment by wigra6** [comment]
Worth mentioning : there is NO SUCH ISSUE IN ORIGINAL YT APP !
Despite YT made changes in video quality settings lately - there is no problem with chosing the desired quality once and for all there.
The reset of user chosen values of video quality shows up in YT Vanced ONLY !

my phone : Xiaomi Redmi Note 8T
Android 10
variant : non-root

**Comment by VancedOfficial** [comment]
Found the issue and it'll be fixed in the next update. In the meantime, change your Vanced override qualities to "Auto" to avoid them being set to the highest resolution possible (Google reversed the quality list so searching the max resolution results in the highest resolution always).

**Comment by wigra6** [comment]
The same problem described here :
[https://github.com/YTVanced/Vanced/issues/690](url)

**Comment by wigra6** [comment]
@VancedOfficial 
No way, in my case AUTO renders the highest available resolution as well !

**Comment by wigra6** [comment]
@VancedOfficial
here is the evidence :
[https://i.imgur.com/PDsiREE.png](url)

**Comment by tanmaykhandual** [comment]
> I am facing the same problem as the "Video quality preference" and "Video settings" in Vanced settings doesn't match.. I've set default resolution for both wifi and mobile to 360p. Then in Video quality preference switched to data saver.. still every time I play a video it resolution is caping at either 1080p or above.. I've just updated to the latest version yesterday..
> 
> Device
> Realme 6, Non Root, Android 11
> Vanced version: 16.14.34

Does the problem resolved

**Comment by Jarvis-Ank** [comment]
> > I am facing the same problem as the "Video quality preference" and "Video settings" in Vanced settings doesn't match.. I've set default resolution for both wifi and mobile to 360p. Then in Video quality preference switched to data saver.. still every time I play a video it resolution is caping at either 1080p or above.. I've just updated to the latest version yesterday..
> > Device
> > Realme 6, Non Root, Android 11
> > Vanced version: 16.14.34
> 
> Does the problem resolved



> > I am facing the same problem as the "Video quality preference" and "Video settings" in Vanced settings doesn't match.. I've set default resolution for both wifi and mobile to 360p. Then in Video quality preference switched to data saver.. still every time I play a video it resolution is caping at either 1080p or above.. I've just updated to the latest version yesterday..
> > Device
> > Realme 6, Non Root, Android 11
> > Vanced version: 16.14.34
> 
> Does the problem resolved

No. I just downgraded the app

**Comment by VancedOfficial** [comment]
> @VancedOfficial
> No way, in my case AUTO renders the highest available resolution as well !

I mean "Auto" from the Vanced settings. When you set it to auto there, the resolution will be what YouTube sets it to and Vanced won't change it. If it still sets it to highest after that, it's a YouTube problem, not Vancced.

**Comment by KevinX8** [comment]
fixed in 16.16.38
